README.md

```diff
@@ -123,6 +123,31 @@ baseImageOverrides:
   github.com/my-user/my-repo/cmd/foo: registry.example.com/base/for/foo
 ```
 
+### Overriding Go build settings
+
+By default, `ko` builds the binary with no additional arguments other than `--trimpath` (if available) and the ones required to name the binary accordingly. You can extend the `go build` arguments by providing extra build flags using a [GoReleaser](https://github.com/goreleaser/goreleaser) influenced `builds` configuration section in your `.ko.yaml`.
+
+```yaml
+builds:
+- id: foo
+  dir: ./foobar/foo
+  flags:
+  - -tags
+  - netgo
+  ldflags:
+  - -s -w
+  - -extldflags "-static"
+- id: bar
+  main: ./foobar/bar/main.go
+  ldflags:
+    - -s
+    - -w
+```
+
+For the build, `ko` will pick the entry based on the respective import path being used. It will be matched against the local path that is configured using `dir` and `main`. In the context of `ko`, it is fine just to specify `dir` with the intended import path.
```

Comment:
Can this work for `ko` builds outside of the current directory?

Reply:
Just to be on the safe side that I do not misunderstand the question, the setup would be something like the following:
```yaml
---
defaultBaseImage: registry.access.redhat.com/ubi8-minimal:latest

builds:
- dir: github.com/shipwright-io/build/cmd/manager
  flags: [ "-tags", "netgo" ]
  ldflags:
  - -s -w
  - -extldflags "-static"
```

```plain
/tmp $ KO_DOCKER_REPO=ko.local go run $GOPATH/src/github.com/google/ko/main.go publish --bare github.com/shipwright-io/build/cmd/manager
2021/04/22 22:59:39 adding import path github.com/shipwright-io/build/cmd/manager to build config overrides
2021/04/22 22:59:39 Using base registry.access.redhat.com/ubi8-minimal:latest for github.com/shipwright-io/build/cmd/manager
2021/04/22 22:59:39 No matching credentials were found for "registry.access.redhat.com/ubi8-minimal", falling back on anonymous
2021/04/22 22:59:40 go build command: /usr/local/bin/go build -tags netgo -ldflags=-s -w -extldflags "-static" -o /var/folders/dc/1k_907394hx5mft__7g269vh0000gn/T/ko458001556/out -trimpath github.com/shipwright-io/build/cmd/manager
2021/04/22 22:59:40 Building github.com/shipwright-io/build/cmd/manager for linux/amd64
2021/04/22 22:59:44 Loading ko.local:1881b84bc433524bed5364e6979f124aca4f6c9663b229baf68f97b06cd9f4c5
2021/04/22 22:59:47 Loaded ko.local:1881b84bc433524bed5364e6979f124aca4f6c9663b229baf68f97b06cd9f4c5
2021/04/22 22:59:47 Adding tag latest
2021/04/22 22:59:47 Added tag latest
ko.local:1881b84bc433524bed5364e6979f124aca4f6c9663b229baf68f97b06cd9f4c5
```

It is currently working, because the key used in the build configuration override map is the qualified import path. I figure I will add an `IsLocalImport` call and the documentation needs to updated to be more precise.

Reply:
Yep, you got it. Thanks!